protos/flyteidl/plugins/ray.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package flyteidl.plugins;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins";
+
+import "google/protobuf/timestamp.proto";
+
+// RayJobSpec defines the desired state of RayJob
+message RayJob {
+  // RayClusterSpec is the cluster template to run the job
+  RayCluster ray_cluster = 1;
+  // runtime_env is base64 encoded.
+  // Ray runtime environments: https://docs.ray.io/en/latest/ray-core/handling-dependencies.html#runtime-environments
+  string runtime_env = 2;
+  // shutdown_after_job_finishes will determine whether to delete the ray cluster once rayJob succeed or failed.
+  bool shutdown_after_job_finishes = 3;
+  // ttl_seconds_after_finished is the TTL to clean up RayCluster.
+  // It's only working when ShutdownAfterJobFinishes set to true.
+  int32 ttl_seconds_after_finished = 4;
+}
+
+// Define Ray cluster spec
+message RayCluster {
+  // Required field. This field indicates ray cluster configuration
+  ClusterSpec cluster_spec = 1;
+
+}
+
+message ClusterSpec {
+  // The head group configuration
+  HeadGroupSpec head_group_spec = 1;
+  // The worker group configurations
+  repeated WorkerGroupSpec worker_group_spec = 2;
+}
+
+message HeadGroupSpec {
+  // Optional. The ray start params of head node group
+  map<string, string> ray_start_params = 1;
+}
+
+message WorkerGroupSpec {
```

Comment:
Could we also add `service_account_name` to `HeadGroupSpec` and `WorkerGroupSpec`? This is useful for folks when they use a specific service account with necessary permissions configured to access cloud resources.

Reply:
We can configure the service account or IAM role on flyteconsole or during registration.
`pyflyte register --service-account default ...`

<img width="849" alt="Screen Shot 2022-07-29 at 3 10 19 AM" src="https://user-images.githubusercontent.com/37936015/181618814-6deda21f-a622-4dba-b6b2-735c89428d47.png">

Reply:
Actually, some task level configs (like resource, secret, service account, IAM) we already have in Flyte. Therefore, we don't need to define it here. https://github.com/flyteorg/flyteidl/blob/e749b82a70027d3364f4e5b98ce2cfbffb4eef54/protos/flyteidl/admin/launch_plan.proto#L96-L107

Reply:
@daikeshi does this sound reasonable to you?